### ISSUE TITLE
regression in service module for docker daemon between 1.8.4 and 1.9.1

### ISSUE THREAD

**Issue by kbroughton** [ansible/ansible-modules-core#1751]
Issue Type:

Bug

Ansible Version:

The following play works in 1.8.4, but not 1.9.1
- name: restart docker
  service: name=docker state=restarted

Ansible Configuration:

No relevant diffs

Environment:

Running from macOSx 10.10.3

Summary:

Regression in service module for docker daemon between 1.8.4 and 1.9.1 possibly due to latest docker 1.7.1 release.

Steps To Reproduce:
# install docker
- name: install docker
  shell: curl -sSL https://get.docker.io/ubuntu/ | sudo sh
# configure docker
- name: change the default group to docker for ubuntu user id
  user: name=ubuntu group=docker
- name: update docker config
  copy: src=docker dest=/etc/default/docker owner=root group=root mode=0644
- name: restart docker
  service: name=docker state=restarted

Expected Results:

I expect the following command to work for all releases of ansible after a certain minimum version.
Perhaps this should be added to regression tests.
- name: restart docker
  service: name=docker state=restarted

Note that this only came up recently and there was just a new release of docker (1.7.1) last week, so this could be a result of a new docker daemon incompatible with a previous version of ansible service module.

Actual Results:
- name: restart docker
  service: name=docker state=restarted

Fails with one of the following messages (ran on colleagues machine. exact output unavailable)

```
    if self.svc_cmd is None and not self.svc_initscript:
        if self.module.params['must_exist']:
            self.module.fail_json(msg='cannot find \'service\' binary or init script for service,  possible typo in service name?, aborting')
        else:

    if self.enable_cmd is None:
        self.module.fail_json(msg='cannot detect command to enable service %s, typo or init system potentially unknown' % self.name)
```

**Comment by abadger** [comment]
This looks like a generic issue with service rather than something specific to docker.  There was an issue that was fixed in 1.9.2 for services that have both an upstart-specific/systemd-specific service description file and a sysv init script:

> - Fixed a bug in the service module where init scripts were being incorrectly used instead of upstart/systemd.
>   - https://github.com/ansible/ansible/blob/devel/CHANGELOG.md

Could you have your colleague upgrade to 1.9.2 and see if that fixes the problem?

**Comment by kbroughton** [comment]
yes, that was the problem.
